clang/lib/DPCT/DPCT.cpp

```diff
@@ -1224,14 +1224,13 @@ int runDPCT(int argc, const char **argv) {
     // /p AnalysisScope defaults to the value of /p InRoot
     AnalysisScope.setValue(InRoot.getValue(), true);
   } else {
-    bool validAnalysisScope =
-        makeCanonical(AnalysisScope) &&
-        isChildOrSamePath(AnalysisScope, InRoot);
-    // /p AnalysisScope must be the same as or parent of /p InRoot
-    if (!validAnalysisScope) {
+    // /p InRoot must be the same as or child of /p AnalysisScope
+    if (!makeCanonical(AnalysisScope) ||
+        !isChildOrSamePath(AnalysisScope, InRoot)) {
       ShowStatus(MigrationErrorInvalidAnalysisScope);
       dpctExit(MigrationErrorInvalidAnalysisScope);
     }
+    ValidateInputDirectory(Tool, AnalysisScope);
```

Comment:
validate AnalysisScope if it's specified by users.
like InRoot, AnalysisScope should not contain CUDA path or dpct tool path.